pkg/engine/variables/operator/numeric.go

```diff
@@ -171,6 +156,11 @@ func (noh NumericOperatorHandler) validateValueWithStringPattern(key string, val
 	if err == nil {
 		return compareByCondition(float64(durationKey.Seconds()), float64(durationValue.Seconds()), noh.condition, &noh.log)
 	}
+	// attempt to extract resource quantity from string before parsing floats/ints as resources can also be ints/floats represented as string type
+	resourceKey, resourceValue, err := parseQuantity(key, value)
+	if err == nil {
+		return compareByCondition(float64(resourceKey.Cmp(resourceValue)), 0, noh.condition, &noh.log)
```

Comment:
I know this line was there before but splitting it into multiple lines would make it much more readable.